src/status_im2/contexts/syncing/events.cljs

```diff
@@ -52,27 +53,35 @@
 (rf/defn initiate-local-pairing-with-connection-string
   {:events       [:syncing/input-connection-string-for-bootstrapping]
    :interceptors [(re-frame/inject-cofx :random-guid-generator)]}
-  [{:keys [random-guid-generator db]} connection-string]
+  [{:keys [random-guid-generator db] :as cofx} connection-string]
   (let [installation-id (random-guid-generator)
         default-node-config (get-default-node-config installation-id)
         default-node-config-string (.stringify js/JSON (clj->js default-node-config))
         callback
         (fn [final-node-config]
-          (let [config-map (.stringify js/JSON
-                                       (clj->js
-                                        {:receiverConfig {:kdfIterations config/default-kdf-iterations
-                                                          :nodeConfig final-node-config
-                                                          :settingCurrentNetwork config/default-network
-                                                          :deviceType platform/os
-                                                          :deviceName
-                                                          (native-module/get-installation-name)}}))]
+          (let [config-map      (.stringify js/JSON
+                                            (clj->js
+                                             {:receiverConfig
+                                              {:kdfIterations config/default-kdf-iterations
+                                               :nodeConfig final-node-config
+                                               :settingCurrentNetwork config/default-network
+                                               :deviceType platform/os
+                                               :deviceName
+                                               (native-module/get-installation-name)}}))
+                includes-error? #(string/includes? res "error")]
             (rf/dispatch [:syncing/update-role constants/local-pairing-role-receiver])
             (native-module/input-connection-string-for-bootstrapping
              connection-string
              config-map
-             #(log/info "Initiated local pairing"
-                        {:response %
-                         :event    :syncing/input-connection-string-for-bootstrapping}))))]
+             (fn [res]
+               (signals/handle-local-pairing-signals
```

Comment:
the function name `signals/handle-local-pairing-signals` is only used for handling local pairing signals rather than returning value from `InputConnectionStringForBootstrapping` , i'm very struggling for such usage. 🥹

Reply:
I agree, Going to refactor it later today.